R/gitlab.R

```diff
@@ -3,11 +3,17 @@ isGitlabURL <- function(url) {
 }
 
 canUseGitlabDownloader <- function() {
-  all(packageVersionInstalled(devtools = "1.9.1", httr = "1.0.0"))
+  (all(packageVersionInstalled(httr = "1.0.0")) &
+     !is.null(gitlab_user(quiet = TRUE)) &
+     !is.null(gitlab_pwd(quiet = TRUE)))
 }
 
 gitlabDownload <- function(url, destfile, ...) {
-  onError(1, gitlabDownloadImpl(url, destfile, ...))
+  tryCatch(
+    gitlabDownloadImpl(url, destfile, ...),
+    error = function(e) {
+      stop(sprintf("GitLab request failed with %s", e), call. = FALSE)
```

Comment:
```suggestion
      stop(sprintf("GitLab request failed with %s", conditionMessage(e)), call. = FALSE)
```

Reply:
As above: message tweaked, but not using `conditionMessage`.